tilgangservice/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/tilgangservice/TilgangOrgService.kt

```diff
@@ -0,0 +1,163 @@
+package no.nav.helsearbeidsgiver.inntektsmelding.tilgangservice
+
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.json.JsonElement
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helsearbeidsgiver.felles.BehovType
+import no.nav.helsearbeidsgiver.felles.EventName
+import no.nav.helsearbeidsgiver.felles.FeilReport
+import no.nav.helsearbeidsgiver.felles.Feilmelding
+import no.nav.helsearbeidsgiver.felles.Key
+import no.nav.helsearbeidsgiver.felles.Tilgang
+import no.nav.helsearbeidsgiver.felles.TilgangData
+import no.nav.helsearbeidsgiver.felles.json.les
+import no.nav.helsearbeidsgiver.felles.json.lesOrNull
+import no.nav.helsearbeidsgiver.felles.json.toJson
+import no.nav.helsearbeidsgiver.felles.json.toMap
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.FailKanal
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.StatefullDataKanal
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.StatefullEventListener
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.composite.CompositeEventListener
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.model.Fail
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.publish
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.redis.RedisKey
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.redis.RedisStore
+import no.nav.helsearbeidsgiver.felles.utils.Log
+import no.nav.helsearbeidsgiver.utils.json.fromJson
+import no.nav.helsearbeidsgiver.utils.json.serializer.UuidSerializer
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import no.nav.helsearbeidsgiver.utils.json.toPretty
+import no.nav.helsearbeidsgiver.utils.log.MdcUtils
+import no.nav.helsearbeidsgiver.utils.log.logger
+import no.nav.helsearbeidsgiver.utils.log.sikkerLogger
+import java.util.UUID
+
+class TilgangOrgService(
+    private val rapid: RapidsConnection,
+    override val redisStore: RedisStore
+) : CompositeEventListener() {
+
+    private val logger = logger()
+    private val sikkerLogger = sikkerLogger()
+
+    override val event = EventName.TILGANG_ORG_REQUESTED
+    override val startKeys = listOf(
+        Key.ORGNRUNDERENHET,
+        Key.FNR
+    )
+    override val dataKeys = listOf(
+        Key.TILGANG
+    )
+
+    init {
+        StatefullEventListener(rapid, event, redisStore, startKeys, ::onPacket)
+        StatefullDataKanal(rapid, event, redisStore, dataKeys, ::onPacket)
+        FailKanal(rapid, event, ::onPacket)
+    }
+
+    override fun new(melding: Map<Key, JsonElement>) {
+        val transaksjonId = Key.UUID.les(UuidSerializer, melding)
+        val orgnr = Key.ORGNRUNDERENHET.les(String.serializer(), melding)
+        val fnr = Key.FNR.les(String.serializer(), melding)
+
+        MdcUtils.withLogFields(
+            Log.klasse(this),
+            Log.event(event),
+            Log.transaksjonId(transaksjonId)
+        ) {
+            rapid.publish(
+                Key.EVENT_NAME to event.toJson(),
```

Comment:
Ikke strengt nødvendig, men kunne kanskje sendt med generert forespørselID til alle løsere / servicer hele veien gjennom i tomtSkjema-løypa også, for å kunne trace hele flyten og mer konsistent oppførsel

Reply:
Jeg tror det fort blir forvirrende om vi sender med en forespørsel-ID for en flyt som ikke har noe med forespørsler å gjøre. Da synes jeg hellre vi burde erstatte forespørsel-ID med en generisk IM-ID som kan brukes for både forespurte og ikke forespurte inntektsmeldinger.